tycho-p2/tycho-p2-plugin/src/main/java/org/eclipse/tycho/plugins/p2/BaselineValidator.java

```diff
@@ -145,8 +148,21 @@ public String getDetailedMessage() {
                             List<Artifact> attachedArtifacts = project.getAttachedArtifacts();
                             ListIterator<Artifact> iterator = attachedArtifacts.listIterator();
                             while (iterator.hasNext()) {
-                                if (classifier.equals(iterator.next().getClassifier())) {
-                                    iterator.remove();
+                                Artifact next = iterator.next();
+                                if (classifier.equals(next.getClassifier())) {
+                                    try {
+                                        iterator.remove();
+                                    } catch (UnsupportedOperationException e) {
+                                        ArrayList<Artifact> list = new ArrayList<>(attachedArtifacts);
+                                        list.remove(next);
+                                        try {
+                                            MethodUtils.invokeMethod(project, true, "setAttachedArtifacts", list);
```

Comment:
Doesn't this cause an illegal reflective access error when the build is performed on a Java-17 VM?

Reply:
Then this workaround won't work of course, but tycho does not work in all combinations of build vm and maven version anyways.

Reply:
But I think the combination of latest Java, latest Maven and latest Tycho is not some special niche configuration but one that should work. Especially because Users are always encouraged to use the latest versions, which I think is good. But then Tycho should support it.

Reply:
One could still use latest java to compile, but feel free to suggest a better alternative here :-)

Reply:
Unfortunately I don't have a better solution immediately at hand.
I know this is not very helpful to find a solution for the moment, but I think it is a point to mention and to consider for the future.

Reply:
This mojo was designed in 2012 so many assumptions of it might not hold true for modern java, at least this allows to use it for some more time until we find a better solution e.g. convince maven to make the method public, provide an alternative approach to remove artifacts or even overhaul the BaseLineVlidator to use a different approach. Even though most eclipse-project use this, I have never seen a use of it before in any other tycho project so maybe this is also a bit eclipse-specific and the default ci-servers use java 11 to build afaik so there is some time to investigate for better solutions until then.